x-pack/plugins/maps/public/classes/styles/vector/properties/dynamic_color_property.test.tsx

```diff
@@ -578,3 +578,44 @@ test('Should read out ordinal type correctly', async () => {
   expect(ordinalColorStyle2.isOrdinal()).toEqual(true);
   expect(ordinalColorStyle2.isCategorical()).toEqual(false);
 });
+
+describe('renderFieldMetaPopover', () => {
+  const nonGeoJsonField = Object.create(mockField);
+  nonGeoJsonField.canReadFromGeoJson = () => {
+    return false;
+  };
+
+  [
+    { field: mockField, name: 'Should enable toggle when field is backed by geojson-source' },
+    {
+      field: nonGeoJsonField,
+      name: 'Should disable toggle when field is not backed by geojson source',
+    },
+  ].forEach((stub) => {
+    test(stub.name, async () => {
+      const colorStyle = makeProperty(
+        {
+          color: 'Blues',
+          type: undefined,
+          fieldMetaOptions,
+        },
+        undefined,
+        stub.field
+      );
+
+      const legendRow = colorStyle.renderFieldMetaPopover(() => {});
+      if (legendRow === null) {
+        throw new Error('Should render');
+      }
+
+      const component = shallow(legendRow);
+
+      // Ensure all promises resolve
```

Comment:
There is nothing async in renderFieldMetaPopover so this async logic is not needed.